doc/themes/the-things-stack/assets/js/theme.js

```diff
@@ -24,4 +24,65 @@ document.addEventListener('DOMContentLoaded', function () {
 			});
 		});
 	}
+
+  const tabs = [...document.querySelectorAll('#tabs li')];
+  const tab_content = [...document.querySelectorAll('#tab-content section')];
+  const active_class = 'is-active';
+
+  function init() {
+    if(window.sessionStorage.getItem('tabActive')) {
+      toggleTab(window.sessionStorage.getItem('tabActive'))
+    }
+
+    tabs.forEach((tab) => {
+      tab.addEventListener('click', handleClick)
+    })
+  }
+
+  function handleClick(event) {
+    event.preventDefault();
+    let tab = event.currentTarget;
+    let key = tab.getAttribute('data-tab');
+    toggleTab(key)
+  }
+
+  function toggleTab(key) {
+    let tab = getTabByKey(key)
+
+    if(!tab) return
+
+    updateActiveTab(tab)
+    updateActiveContent(key)
+    if(window.sessionStorage){
+      window.sessionStorage.setItem("tabActive", key)
+    }
+  }
+
+  function getTabByKey(key) {
+    return document.querySelectorAll(`[data-tab="${key}"]`)[0];
+  }
+
+  function updateActiveTab(tab) {
+    tabs.forEach((tab) => {
+      if (tab && tab.classList.contains(active_class)) {
+        tab.classList.remove(active_class);
+      }
+    });
+    tab.classList.add(active_class);
+  }
+
+  function updateActiveContent(key) {
+    tab_content.forEach((item) => {
+      if (item && item.classList.contains(active_class)) {
+        item.classList.remove(active_class);
+      }
+      let data = item.getAttribute('data-content');
+      if (data === key) {
+        item.classList.add(active_class);
+      }
+    });
+  }
```

Comment:
You can merge these as `setActiveTab(key)` since they comprise the same concern and as far as I see, you wouldn't call one without the other.